lib/membrane_opus/parser.ex

```diff
@@ -1,79 +1,210 @@
+        Input caps delimitiation: #{self_delimiting?}
+        Requested delimitation: #{state.delimitation}
+        """
+    end
+  end
+
+  @spec delimit(binary, [non_neg_integer], pos_integer) :: binary
+  defp delimit(data, frame_lengths, header_size) do
+    <<head::binary-size(header_size), body::binary>> = data
+
+    [
+      head,
+      frame_lengths |> List.last() |> FrameLengths.encode_frame_length(),
+      body
+    ]
+    |> :binary.list_to_bin()
+  end
+
+  @spec undelimit(binary, [non_neg_integer], pos_integer) :: binary
+  defp undelimit(data, frame_lengths, header_size) do
+    last_length = frame_lengths |> List.last() |> FrameLengths.encode_frame_length()
+    last_length_size = byte_size(last_length)
+    parsed_header_size = header_size - last_length_size
+
+    <<parsed_head::binary-size(parsed_header_size), _last_length::binary-size(last_length_size),
+      body::binary>> = data
+
+    [
+      parsed_head,
+      body
+    ]
+    |> :binary.list_to_bin()
+  end
+
+  # parses config number, stereo flag, and frame packing strategy from the TOC
+  # byte
+  @spec parse_toc_byte(binary) :: {0..31, 0..1, 0..3, binary}
+  defp parse_toc_byte(data) do
+    <<configuration_number::size(5), stereo_flag::size(1), frame_packing::size(2), rest::binary>> =
+      data
+
+    {configuration_number, stereo_flag, frame_packing, rest}
+  end
+
+  # parses configuration values from TOC configuration number
+  @spec parse_configuration(0..31) ::
+          {:silk | :hybrid | :celt, :narrow | :medium | :wide | :super_wide | :full,
+           float | 5 | 10 | 20 | 40 | 60}
+  defp parse_configuration(configuration_number) do
+    case configuration_number do
+      0 -> {:silk, :narrow, 10}
+      1 -> {:silk, :narrow, 20}
+      2 -> {:silk, :narrow, 40}
+      3 -> {:silk, :narrow, 60}
+      4 -> {:silk, :medium, 10}
+      5 -> {:silk, :medium, 20}
+      6 -> {:silk, :medium, 40}
+      7 -> {:silk, :medium, 60}
+      8 -> {:silk, :wide, 10}
+      9 -> {:silk, :wide, 20}
+      10 -> {:silk, :wide, 40}
+      11 -> {:silk, :wide, 60}
+      12 -> {:hybrid, :super_wide, 10}
+      13 -> {:hybrid, :super_wide, 20}
+      14 -> {:hybrid, :full, 10}
+      15 -> {:hybrid, :full, 20}
+      16 -> {:celt, :narrow, 2.5}
+      17 -> {:celt, :narrow, 5}
+      18 -> {:celt, :narrow, 10}
+      19 -> {:celt, :narrow, 20}
+      20 -> {:celt, :wide, 2.5}
+      21 -> {:celt, :wide, 5}
+      22 -> {:celt, :wide, 10}
+      23 -> {:celt, :wide, 20}
+      24 -> {:celt, :super_wide, 2.5}
+      25 -> {:celt, :super_wide, 5}
+      26 -> {:celt, :super_wide, 10}
+      27 -> {:celt, :super_wide, 20}
+      28 -> {:celt, :full, 2.5}
+      29 -> {:celt, :full, 5}
+      30 -> {:celt, :full, 10}
+      31 -> {:celt, :full, 20}
     end
   end
+
+  # determines number of channels
+  @spec parse_channels(0..1) :: 1..2
+  defp parse_channels(stereo_flag) when stereo_flag in 0..1, do: stereo_flag + 1
```

Comment:
I'd move these to a separate module. Maybe to FrameLengths, and rename it accordingly?